File: MyClub/MyClub.Model/Requests/UserUpsertRequest.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MyClub.Model.Requests
{
    public class UserUpsertRequest
    {
        [Required]
        [StringLength(50)]
        public string FirstName { get; set; } = string.Empty;
        
        [Required]
        [StringLength(50)]
        public string LastName { get; set; } = string.Empty;
        
        [Required]
        [StringLength(100)]
        public string Username { get; set; } = string.Empty;
        
        [Required]
        [StringLength(100)]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;
        
        [StringLength(20)]
        [Phone]
        public string? PhoneNumber { get; set; }
        
        public bool IsActive { get; set; } = true;
        
        // Password is only required for new users
        [StringLength(100, MinimumLength = 6)]
        public string? Password { get; set; }
        
        // Role ID (null means keep existing role or use default)
        public int? RoleId { get; set; }
    }
} 

File: MyClub/UI/myclub_mobile/lib/models/requests/user_upsert_request.dart
/// User registration/update request model
class UserUpsertRequest {
  final String firstName;
  final String lastName;
  final String username;
  final String email;
  final String? phoneNumber;
  final bool isActive;
  final String? password;
  final int? roleId;

  UserUpsertRequest({
    required this.firstName,
    required this.lastName,
    required this.username,
    required this.email,
    this.phoneNumber,
    this.isActive = true,
    this.password,
    this.roleId,
  });

  /// Convert to JSON for API requests
  Map<String, dynamic> toJson() {
    return {
      'firstName': firstName,
      'lastName': lastName,
      'username': username,
      'email': email,
      if (phoneNumber != null) 'phoneNumber': phoneNumber,
      'isActive': isActive,
      if (password != null) 'password': password,
      if (roleId != null) 'roleId': roleId,
    };
  }

  /// Create from JSON response
  factory UserUpsertRequest.fromJson(Map<String, dynamic> json) {
    return UserUpsertRequest(
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      phoneNumber: json['phoneNumber'],
      isActive: json['isActive'] ?? true,
      password: json['password'],
      roleId: json['roleId'],
    );
  }
}
